### ISSUE TITLE
conjure-up: No controllers registered

### ISSUE THREAD

**Issue by panaceya** [conjure-up/spells#69]
Hi!

I use [this](https://www.ubuntu.com/download/cloud/autopilot) manual to startup OpenStack. 

On stage 6 after run command `conjure-up --bootstrap-to simple-collie.maas` and push on button "Deploy" i see next trace:

```
Exception in ev.run():
Traceback (most recent call last):
  File "/snap/conjure-up/352/lib/python3.6/site-packages/ubuntui/ev.py", line 82, in run
    cls.loop.run()
  File "/snap/conjure-up/352/lib/python3.6/site-packages/urwid/main_loop.py", line 278, in run
    self._run()
  File "/snap/conjure-up/352/lib/python3.6/site-packages/urwid/main_loop.py", line 376, in _run
    self.event_loop.run()
  File "/snap/conjure-up/352/lib/python3.6/site-packages/urwid/main_loop.py", line 1328, in run
    raise self._exc_info[0](self._exc_info[1]).with_traceback(self._exc_info[2])
  File "/snap/conjure-up/352/usr/lib/python3.6/asyncio/events.py", line 126, in _run
    self._callback(*self._args)
  File "/snap/conjure-up/352/lib/python3.6/site-packages/conjureup/ui/views/deploystatus.py", line 31, in _refresh_nodes_on_main_thread
    status = model_status()
  File "/snap/conjure-up/352/lib/python3.6/site-packages/conjureup/juju.py", line 32, in _decorator
    login(force=True)
  File "/snap/conjure-up/352/lib/python3.6/site-packages/conjureup/juju.py", line 123, in login
    uuid = get_model(app.current_controller, app.current_model)['model-uuid']
  File "/snap/conjure-up/352/lib/python3.6/site-packages/conjureup/juju.py", line 787, in get_model
    models = get_models(controller)['models']
  File "/snap/conjure-up/352/lib/python3.6/site-packages/conjureup/juju.py", line 858, in get_models
    "Unable to list models: {}".format(sh.stderr.decode('utf8')))
LookupError: Unable to list models: error: No controllers registered.

Please either create a new controller using "juju bootstrap" or connect to
another controller that you have been given access to using "juju register".


Traceback (most recent call last):
  File "/snap/conjure-up/352/bin/conjure-up", line 11, in <module>
    load_entry_point('conjure-up==2.1.5', 'console_scripts', 'conjure-up')()
  File "/snap/conjure-up/352/lib/python3.6/site-packages/conjureup/app.py", line 369, in main
    EventLoop.run()
  File "/snap/conjure-up/352/lib/python3.6/site-packages/ubuntui/ev.py", line 82, in run
    cls.loop.run()
  File "/snap/conjure-up/352/lib/python3.6/site-packages/urwid/main_loop.py", line 278, in run
    self._run()
  File "/snap/conjure-up/352/lib/python3.6/site-packages/urwid/main_loop.py", line 376, in _run
    self.event_loop.run()
  File "/snap/conjure-up/352/lib/python3.6/site-packages/urwid/main_loop.py", line 1328, in run
    raise self._exc_info[0](self._exc_info[1]).with_traceback(self._exc_info[2])
  File "/snap/conjure-up/352/usr/lib/python3.6/asyncio/events.py", line 126, in _run
    self._callback(*self._args)
  File "/snap/conjure-up/352/lib/python3.6/site-packages/conjureup/ui/views/deploystatus.py", line 31, in _refresh_nodes_on_main_thread
    status = model_status()
  File "/snap/conjure-up/352/lib/python3.6/site-packages/conjureup/juju.py", line 32, in _decorator
    login(force=True)
  File "/snap/conjure-up/352/lib/python3.6/site-packages/conjureup/juju.py", line 123, in login
    uuid = get_model(app.current_controller, app.current_model)['model-uuid']
  File "/snap/conjure-up/352/lib/python3.6/site-packages/conjureup/juju.py", line 787, in get_model
    models = get_models(controller)['models']
  File "/snap/conjure-up/352/lib/python3.6/site-packages/conjureup/juju.py", line 858, in get_models
    "Unable to list models: {}".format(sh.stderr.decode('utf8')))
LookupError: Unable to list models: error: No controllers registered.

Please either create a new controller using "juju bootstrap" or connect to
another controller that you have been given access to using "juju register".


exception calling callback for <Future at 0x7f440a5b7240 state=finished returned NoneType>
Traceback (most recent call last):
  File "/snap/conjure-up/352/usr/lib/python3.6/concurrent/futures/_base.py", line 297, in _invoke_callbacks
    callback(self)
  File "/snap/conjure-up/352/lib/python3.6/site-packages/conjureup/controllers/deploystatus/gui.py", line 30, in __wait_for_applications
    relations_done_future.result()
AttributeError: 'NoneType' object has no attribute 'result'
```
cat ~/.cache/conjure-up/conjure-up.log      
```
2017-06-14 10:40:41,594 [DEBUG] log.py:14 - conjure-up/_unspecified_spell: LXD version: 2.14, Juju version: 2.1.3-xenial-amd64
2017-06-14 10:40:52,819 [DEBUG] log.py:14 - conjure-up/_unspecified_spell: Path is local filesystem, copying /snap/conjure-up/352/spells/landscape to /home/panaceya/.cache/conjure-up/landscape
2017-06-14 10:40:52,892 [DEBUG] log.py:14 - conjure-up/_unspecified_spell: Pulling bundle for landscape-dense-maas from channel: stable
2017-06-14 10:41:06,797 [DEBUG] log.py:14 - conjure-up/_unspecified_spell: Performing bootstrap: conjure-up-maas-b1a maas/1.1.130.2
2017-06-14 10:41:06,798 [DEBUG] log.py:14 - conjure-up/_unspecified_spell: bootstrap cmd: juju bootstrap maas/1.1.130.2 conjure-up-maas-b1a --config image-stream=daily --config enable-os-upgrade=false --default-model conjure-landscape-117 --to simple-collie.maas --bootstrap-series=xenial --credential conjure-up-maas-073 --debug
2017-06-14 10:41:07,300 [DEBUG] log.py:14 - conjure-up/_unspecified_spell: Failed to find cloud in list-clouds, attempting to read bootstrap-config
2017-06-14 10:41:08,528 [DEBUG] log.py:14 - conjure-up/_unspecified_spell: Failed to find cloud in list-clouds, attempting to read bootstrap-config
2017-06-14 10:41:09,207 [DEBUG] log.py:14 - conjure-up/_unspecified_spell: Rendering bootstrap wait
2017-06-14 10:41:09,763 [DEBUG] log.py:14 - conjure-up/_unspecified_spell: Failed to find cloud in list-clouds, attempting to read bootstrap-config
2017-06-14 10:41:11,007 [DEBUG] log.py:14 - conjure-up/_unspecified_spell: Failed to find cloud in list-clouds, attempting to read bootstrap-config
2017-06-14 10:41:12,285 [DEBUG] log.py:14 - conjure-up/_unspecified_spell: Failed to find cloud in list-clouds, attempting to read bootstrap-config
2017-06-14 10:41:13,539 [DEBUG] log.py:14 - conjure-up/_unspecified_spell: Failed to find cloud in list-clouds, attempting to read bootstrap-config
2017-06-14 10:41:17,403 [DEBUG] log.py:14 - conjure-up/_unspecified_spell: Failed to find cloud in list-clouds, attempting to read bootstrap-config
2017-06-14 10:41:18,619 [DEBUG] log.py:14 - conjure-up/_unspecified_spell: Failed to find cloud in list-clouds, attempting to read bootstrap-config
2017-06-14 10:41:19,844 [DEBUG] log.py:14 - conjure-up/_unspecified_spell: Failed to find cloud in list-clouds, attempting to read bootstrap-config
2017-06-14 10:41:21,103 [DEBUG] log.py:14 - conjure-up/_unspecified_spell: Failed to find cloud in list-clouds, attempting to read bootstrap-config
2017-06-14 10:41:24,973 [DEBUG] log.py:14 - conjure-up/_unspecified_spell: handle bootstrap
2017-06-14 10:41:24,974 [ERROR] log.py:17 - conjure-up/_unspecified_spell: 10:41:06 INFO  juju.cmd supercommand.go:63 running juju [2.1.3 gc go1.6.2]

10:41:06 DEBUG juju.cmd supercommand.go:64   args: []string{"juju", "bootstrap", "maas/1.1.130.2", "conjure-up-maas-b1a", "--config", "image-stream=daily", "--config", "enable-os-upgrade=false", "--default-model", "conjure-landscape-117", "--to", "simple-collie.maas", "--bootstrap-series=xenial", "--credential", "conjure-up-maas-073", "--debug"}

10:41:06 INFO  cmd cmd.go:141 cloud "maas" not found, trying as a provider name

10:41:06 INFO  cmd cmd.go:141 interpreting "1.1.130.2" as the cloud endpoint

10:41:06 DEBUG juju.cmd.juju.commands bootstrap.go:780 authenticating with region "" and credential "conjure-up-maas-073" ()

10:41:06 DEBUG juju.cmd.juju.commands bootstrap.go:892 provider attrs: map[]

10:41:07 INFO  cmd cmd.go:141 Adding contents of "/home/panaceya/.local/share/juju/ssh/juju_id_rsa.pub" to authorized-keys

10:41:07 DEBUG juju.cmd.juju.commands bootstrap.go:948 preparing controller with config: map[enable-os-refresh-update:true disable-network-management:false apt-https-proxy: https-proxy: type:maas default-series:xenial uuid:b0bfb494-24c2-45dd-855f-896e5deeacc1 logging-config: image-stream:daily no-proxy: proxy-ssh:false firewall-mode:instance transmit-vendor-metrics:true net-bond-reconfigure-delay:17 authorized-keys:ssh-rsa AAAAB3xxxHIDDENxxx3 juju-client-key

 apt-ftp-proxy: apt-mirror: test-mode:false ftp-proxy: ssl-hostname-verification:true ignore-machine-addresses:false name:controller automatically-retry-hooks:true logforward-enabled:false development:false apt-http-proxy: enable-os-upgrade:false agent-metadata-url: provisioner-harvest-mode:destroyed http-proxy: agent-stream:released image-metadata-url: resource-tags:]

10:41:07 DEBUG juju.provider.maas environprovider.go:51 opening model "controller".

10:41:20 INFO  cmd cmd.go:129 Creating Juju controller "conjure-up-maas-b1a" on maas

10:41:21 INFO  juju.cmd.juju.commands bootstrap.go:526 combined bootstrap constraints: 

10:41:21 DEBUG juju.environs.bootstrap bootstrap.go:199 model "controller" supports service/machine networks: true

10:41:21 DEBUG juju.environs.bootstrap bootstrap.go:201 network management by juju enabled: true

10:41:21 INFO  cmd cmd.go:141 Loading image metadata

10:41:21 INFO  cmd cmd.go:129 Looking for packaged Juju agent version 2.1.3 for amd64

10:41:21 INFO  juju.environs.bootstrap tools.go:72 looking for bootstrap agent binaries: version=2.1.3

10:41:21 INFO  juju.environs.tools tools.go:101 finding agent binaries in stream "released"

10:41:21 INFO  juju.environs.tools tools.go:103 reading agent binaries with major.minor version 2.1

10:41:21 INFO  juju.environs.tools tools.go:111 filtering agent binaries by version: 2.1.3

10:41:21 INFO  juju.environs.tools tools.go:114 filtering agent binaries by series: xenial

10:41:21 INFO  juju.environs.tools tools.go:117 filtering agent binaries by architecture: amd64

10:41:21 DEBUG juju.environs.tools urls.go:109 trying datasource "keystone catalog"

10:41:21 DEBUG juju.environs.simplestreams simplestreams.go:680 using default candidate for content id "com.ubuntu.juju:released:tools" are {20161007 mirrors:1.0 content-download streams/v1/cpc-mirrors.sjson []}

10:41:22 INFO  juju.environs.bootstrap tools.go:74 found 1 packaged agent binaries

10:41:23 INFO  cmd cmd.go:141 Starting new instance for initial controller

Launching controller instance(s) on maas...

10:41:24 INFO  juju.provider.common destroy.go:20 destroying model "controller"

10:41:24 INFO  juju.provider.common destroy.go:31 destroying instances

10:41:24 INFO  juju.provider.common destroy.go:51 destroying storage

10:41:24 ERROR cmd supercommand.go:458 failed to bootstrap model: cannot start bootstrap instance: cannot run instances: cannot run instance: No available machine matches constraints: mem=3584.0 name=simple-collie.maas

10:41:24 DEBUG cmd supercommand.go:459 (error details: [{github.com/juju/juju/cmd/juju/commands/bootstrap.go:574: failed to bootstrap model} {github.com/juju/juju/provider/common/bootstrap.go:179: cannot start bootstrap instance} {github.com/juju/juju/provider/maas/environ.go:950: cannot run instances} {github.com/juju/juju/provider/maas/environ.go:1310: cannot run instance} {github.com/juju/juju/provider/maas/environ.go:751: } {github.com/juju/gomaasapi/controller.go:511: } {github.com/juju/gomaasapi/controller.go:708: } {github.com/juju/gomaasapi/controller.go:739: } {github.com/juju/gomaasapi/client.go:123: } {ServerError: 409 CONFLICT (No available machine matches constraints: mem=3584.0 name=simple-collie.maas)}])

2017-06-14 10:41:24,977 [DEBUG] log.py:14 - conjure-up/_unspecified_spell: Showing dialog for exception: Juju failed to bootstrap: maas
```

**Comment by battlemidget** [comment]
Can you try with

```
sudo snap refresh conjure-up --candidate
```

**Comment by battlemidget** [comment]
Oh this is an error with MAAS and not conjure-up

```
10:41:24 ERROR cmd supercommand.go:458 failed to bootstrap model: cannot start bootstrap instance: cannot run instances: cannot run instance: No available machine matches constraints: mem=3584.0 name=simple-collie.maas

10:41:24 DEBUG cmd supercommand.go:459 (error details: [{github.com/juju/juju/cmd/juju/commands/bootstrap.go:574: failed to bootstrap model} {github.com/juju/juju/provider/common/bootstrap.go:179: cannot start bootstrap instance} {github.com/juju/juju/provider/maas/environ.go:950: cannot run instances} {github.com/juju/juju/provider/maas/environ.go:1310: cannot run instance} {github.com/juju/juju/provider/maas/environ.go:751: } {github.com/juju/gomaasapi/controller.go:511: } {github.com/juju/gomaasapi/controller.go:708: } {github.com/juju/gomaasapi/controller.go:739: } {github.com/juju/gomaasapi/client.go:123: } {ServerError: 409 CONFLICT (No available machine matches constraints: mem=3584.0 name=simple-collie.maas)}])
```

conjure-up can not power on instances from your MAAS server, please verify that is working prior to running conjure-up.

**Comment by panaceya** [comment]
`sudo snap refresh conjure-up --candidate` not help for me. 

MAAS also available for node.
